transport/conn.go

```diff
@@ -437,3 +447,18 @@ func (c *Conn) Close() {
 func (c *Conn) String() string {
 	return fmt.Sprintf("[addr=%s shard=%d]", c.conn.RemoteAddr(), c.shard)
 }
+
+func (c *Conn) registerEvents(e []frame.EventType) (eventHandler, error) {
```

Comment:
You could make the API nicer like `registerEvents(e... frame.EventType)` it would auto make the slice.